DocumentDataAPI/DocumentDataAPI/Data/Repositories/NpgSimilarDocumentRepository.cs

```diff
@@ -63,11 +68,18 @@ public async Task<IEnumerable<long>> AddBatch(List<SimilarDocumentModel> models)
 
     public async Task<int> Delete(long mainDocumentId, long similarDocumentId)
     {
-        _logger.LogDebug("Deleting Document with id {mainDocumentId} {similarDocumentId} from database", mainDocumentId, similarDocumentId);
+        _logger.LogDebug("Deleting similar document with id {similarDocumentId} from document with id: {mainDocumentId} in the database", similarDocumentId, mainDocumentId);
         using IDbConnection con = _connectionFactory.CreateConnection();
         return await con.ExecuteAsync($"delete from similar_documents where {SimilarDocumentMap.MainDocumentId} = @mainDocumentId and " +
             $"{SimilarDocumentMap.SimilarDocumentId} = @similarDocumentId",
-            new { mainDocumentId = mainDocumentId, similarDocumentId = similarDocumentId });
+            new { mainDocumentId, similarDocumentId });
+    }
+
+    public async Task<int> DeleteAll()
+    {
+        _logger.LogDebug("Deleting all similar document from database");
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.ExecuteAsync($"truncate table similar_documents");
```

Comment:
Det er ikke muligt at truncate med read/write brugeren.

Reply:
Tror også noget af det der blev efterspurgt var en Delete metode der tager imod et main document id - som sletter alle entries for det main document id, ikke hele tabellen

Reply:
Jeg ved heller ikke om det er den rigtige funktionalitet ift. min forrige kommentar. Jeres Delete metode sletter ét similar document fra et main document, men der mangler vel en delete metode, som sletter alle similar documents fra et main document. Men det ved i bedst, hvordan det vil komme til at fungere, når similar documents skal opdateres i systemet. Ville man indsætte nye similar documents på alle documents - for så skal hele table slettes, som i gør - ellers skal i nok bare have en metode som reflektere hvorledes delete actions vil komme til at foregå, når ting skal opdateres.
Husk at endpoints er eternal - altså at i nu definere dem for Knox's fremtid. Derfor skal disse overvejelser gerne gøres så der ikke er "development kode" i knox pipeline.